src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -313,10 +322,76 @@ export function InternalDashboardTopNav({
       });
     }
     if (showWriteControls && managed) {
-      allBadges.push(getManagedContentBadge(dashboardManagedBadge.getTooltip()));
+      const badgeProps = {
+        ...getManagedContentBadge(dashboardManagedBadge.getAriaLabel()),
+        onClick: () => setIsPopoverOpen(!isPopoverOpen),
+        onClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+        iconOnClick: () => setIsPopoverOpen(!isPopoverOpen),
+        iconOnClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+      } as TopNavMenuBadgeProps;
+
+      const maybeRedirect = (result?: SaveDashboardReturn) => {
+        if (!result) return;
+        const { redirectRequired, id } = result;
+        if (redirectRequired) {
+          redirectTo({
+            id,
+            editMode: true,
+            useReplace: true,
+            destination: 'dashboard',
+          });
+        }
+      };
```

Comment:
Sorry, we really try to reduce duplicated code 🙈 Can we move `maybeRedirect` from being defined in `use_dashboard_menu_items` to being defined in `internal_dashboard_top_nav` instead? And then we can pass it in as an argument to `useDashboardMenuItems`.

Reply:
My bad! This makes a lot of sense - updated in [f7d2bd6](https://github.com/elastic/kibana/pull/189404/commits/f7d2bd694033d8197e0b5349c78b266708c96783)